### DISCUSSION TITLE
Is there a way to create a categorical heatmap?

### DISCUSSION THREAD

**Discussion by JoshMcRose** [JetBrains/lets-plot-kotlin#138] [Q&A] [answered]
I want to create something like the bistro.corr graphs, but where two columns in my data frame are defined as x and y and the fill color is defined as the count between them. Is this possible, or is there a way to go about making this work with an existing plot type?

**Comment by alshan** [comment, marked as answer]
Try to use [`geomTile()`](https://lets-plot.org/kotlin/-lets--plot--kotlin/org.jetbrains.letsPlot.geom/geom-tile/index.html).
See an example in cell [4] [here](https://nbviewer.org/github/JetBrains/lets-plot-kotlin/blob/master/docs/examples/jupyter-notebooks/contours.ipynb).

**Reply by JoshMcRose** [reply]
This ended up being exactly the answer. Thank you again!

**Comment by JoshMcRose** [comment]
Hey Alshan,

Thanks for getting back to me!

I've been trying geomTile, but I've run into a few issues with getting it to work. I've come back to it this evening, and I'm getting an interesting error (attached) that I really don't understand.  
![Screenshot 2022-10-25 at 19 45 12](https://user-images.githubusercontent.com/66621641/197834391-7e569c5d-7ca1-4f2a-9ad7-4fb507f91965.png)

A snippet of my code: 
  ```
    val z = data["questionText"].values().zip(data["answerOptions"].values())

    letsPlot(data.toMap()) {
        x = "questionText"
        y = "answerOptions"
        fill = z
    }
        .plus(geomTile())
        .plus(scaleFillHue())
        .show()
```

`data["questionText"]` and `data["answerOptions"]` are both columns of strings. To get my z values, I've zipped them into a list of pairs. This might be where I'm going wrong, but I'm not sure how else to make `fill` represent the count of the relationship between `x` and `y`. 

If you have any advice, I'd be all ears.

**Reply by alshan** [reply]
Interesting issue. Try to add `.map { it.toString() }` to the `z` expression to make it a list of strings.

**Reply by JoshMcRose** [reply]
Like this? : `val z = data["questionText"].values().zip(data["answerOptions"].values()).map { it.toString() }`

It returned the same error.

**Comment by alshan** [comment]
It seems there is an issue related to `geomTile()` and categorical x/y. I'll need to take closer look.

**Reply by JoshMcRose** [reply]
Sure, thank you for looking into it!

**Reply by alshan** [reply]
Hi, it appears I'm not able to reproduce this exact issue :( Could you provide a minimal code to reproduce?

**Reply by JoshMcRose** [reply]
Hey Alshan,

I was putting together my data in redacted form to share with you along with my code, and the issue disappeared. It seems that the the string data in the `"questionText"` and `"answerOptions"` columns was too long. By adding `+ ggsize(width = 1200, height = 750)`, the error went away.

Thank you for the help!

**Reply by alshan** [reply]
I see, I'd been experimenting with tiles+small ggsize too, but in a jupyter notebook which is likely not the setup you have. I'll try in Swing/Batik later.
This issue is something that certainly must be fixed.
Thanx for reporting it!
